### PR TITLE
Print LDAP diagnostic messages on error

### PR DESCRIPTION
**Pull request by tiran**
ipa_ldap_init(), ipa_tls_ssl_init(), and the bind operations of ipa-join
and ipa-getkeytab now print LDAP error string and LDAP diagonstic messages
to stderr.

Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: util/ipa_ldap.c [hunk 1]
```diff
@@ -24,6 +24,35 @@
...
+    ret = ldap_get_option(ld, LDAP_OPT_DIAGNOSTIC_MESSAGE, (void*)&diagnostic);
+    has_diagnostic = ((ret == LDAP_SUCCESS) && diagnostic && *diagnostic);
+
+    if (errstring && has_diagnostic) {
+        fprintf(stderr, "%s: %s\n", errstring, diagnostic);
+    } else if (errstring) {
+        fprintf(stderr, "%s\n", errstring);
+    } else if (has_diagnostic) {
+        fprintf(stderr, "%i: %s", errnum, diagnostic);
+    }
```
Comment (rcritten) [inline util/ipa_ldap.c]: Can you add a doc-string that msg needs to be new-line terminated otherwise the output will be smashed together?

Should the corner case of errstring=NULL and has_diagnostic=0 be handled?

**Comment by tiran** [comment]
Example error message:

```
# ./ipa-getkeytab -p test/invalid -k /tmp/keytab -s untrusted.server.example
Unable to initialize STARTTLS session
Connect error: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed (self signed certificate in certificate chain)
Failed to bind to server!
Retrying with pre-4.0 keytab retrieval method...
Unable to initialize STARTTLS session
Connect error: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed (self signed certificate in certificate chain)
Failed to bind to server!
Failed to get keytab
````

File: util/ipa_ldap.c (hunk 1 above)
Comment (rcritten) [reply util/ipa_ldap.c]: Arguably there is nothing to print in that case. Perhaps I wasn't quite awake yet :/ The docstring might be nice though.

Comment (tiran) [reply util/ipa_ldap.c]: I have added a doc string to the new function.

**Comment by rcritten** [comment]
This LGTM.

**Comment by tiran** [comment]
master:

* e107b8e4fe1f205cc1bf992aae2e46a7ab02ebb6 Print LDAP diagnostic messages on error
